lib/supply/client.rb

```diff
@@ -208,90 +139,62 @@ def apk_listings(apk_version_code)
-      )
-      raise result.error_message.red if result.error?
+      listing = Androidpublisher::Listing.new({
+        language: language,
+        title: title,
+        full_description: full_description,
+        short_description: short_description,
+        video: video
+      })
+
+      android_publisher.update_listing(
+        current_package_name,
+        current_edit.id,
+        language,
+        listing)
     end
 
     def upload_apk(path_to_apk)
       ensure_active_edit!
 
-      apk = Google::APIClient::UploadIO.new(File.expand_path(path_to_apk), 'application/vnd.android.package-archive')
-      result_upload = api_client.execute(
-        api_method: android_publisher.edits.apks.upload,
-        parameters: {
-          'editId' => current_edit.data.id,
-          'packageName' => current_package_name,
-          'uploadType' => 'media'
-        },
-        media: apk,
-        authorization: auth_client
-      )
-
-      raise result_upload.error_message.red if result_upload.error?
-
-      return result_upload.data.versionCode
+      result_upload = android_publisher.upload_apk(
+        current_package_name,
+        current_edit.id,
+        upload_source: path_to_apk)
+
+      return result_upload.version_code
     end
 
     def update_track(track, rollout, apk_version_code)
       ensure_active_edit!
 
-      track_body = {
-        'track' => track,
-        'userFraction' => rollout,
-        'versionCodes' => [apk_version_code]
-      }
-
-      result_update = api_client.execute(
-        api_method: android_publisher.edits.tracks.update,
-        parameters:
-          {
-            'editId' => current_edit.data.id,
-            'packageName' => current_package_name,
-            'track' => track
-          },
-        body_object: track_body,
-        authorization: auth_client)
-
-      raise result_update.error_message.red if result_update.error?
+      track_body = Androidpublisher::Track.new({
+        track: track,
+        user_fraction: rollout,
+        version_codes: [apk_version_code]
+      })
+
+      android_publisher.update_track(
+        current_package_name,
+        current_edit.id,
+        track,
+        track_body)
     end
 
     def update_apk_listing_for_language(apk_listing)
       ensure_active_edit!
 
-      body_object = {
-        'language' => apk_listing.language,
-        'recentChanges' => apk_listing.recent_changes
-      }
-
-      result = api_client.execute(
-        api_method: android_publisher.edits.apklistings.update,
-        parameters: {
-          'apkVersionCode' => apk_listing.apk_version_code,
-          'editId' => current_edit.data.id,
-          'packageName' => current_package_name,
-          'language' => apk_listing.language
-        },
-        body_object: body_object,
-        authorization: auth_client
-      )
-      raise result.error_message.red if result.error?
+      apk_listing_object = Androidpublisher.ApkListing.new({
```

Comment:
I think this should be `apk_listing_object = Androidpublisher::ApkListing.new({` 

:: instead of .